src/vendor/processors/stripe_processor.py

```diff
@@ -1253,3 +1258,35 @@ def subscription_payment(self, subscription):
 
         self.subscription_id = stripe_subscription.id
 
+    def charge_customer_profile(self):
+        invoice_data = self.build_invoice()
+        stripe_invoice = self.stripe_create_object(self.stripe.Invoice, invoice_data)
+        if not stripe_invoice:
+            return None
+
+        stripe_customer_payment_methods = self.get_customer_payment_methods(self.invoice.profile.meta['stripe_id'])
+        if not stripe_customer_payment_methods:
+            return None
+        
+        self.invoice.vendor_notes['stripe_id'] = stripe_invoice.id
+        self.invoice.save()
+        
+        stripe_line_items = []
+        for order_item in self.invoice.get_one_time_transaction_order_items():
+            line_item_data = self.build_invoice_line_item(order_item, stripe_invoice.id)
+            stripe_line_item = self.stripe_create_object(self.stripe.InvoiceItem, line_item_data)
+            stripe_line_items.append(stripe_line_item)
+        
+        stripe_invoice.lines = stripe_line_items
+
+        amount = self.convert_decimal_to_integer(self.invoice.get_one_time_transaction_total())
```

Comment:
just curious why we're converting to integer here- does stripe not deal with cents or something?

Reply:
yes it does but in a weird way